### DISCUSSION TITLE
DAE visualization in FoxGloveStudioWeb

### DISCUSSION THREAD

**Discussion by MarcoGi89** [foxglove/community#737] [Q&A] [answered]
Hi all,

I am trying to visualize my robotic scenario spawning my robot, following a similar approach reported [here](https://foxglove.dev/blog/simulating-robotic-scenarios-with-ros1-and-gazebo)

I have already manage to visualize the complete scenario in rViz, so I connect foxglove_studio web to Rosbridge.
I succeed to see all tfs correctly published but not the DAE of the robot.

I modified the urdf changing the path of the DAE file as shown below:

`<geometry> 
<mesh filename="http://localhost:3000/urdf/ur5e/meshes/base.dae"/>
   </geometry>`

This files are provided by a service I created on port 3000 and files are available, but DAE are not visualized in 3DPanel.

What am I doing wrong?

I am using Ros2 humble with Rosbridge websocket server.

Hope these information are enough. 

Thank you in `advance.`

**Comment by achim-k** [comment]
A few questions:

- Do you see the `/robot_description` topic in Studio? When you open it in a Raw message panel, do you see the correct mesh filenames?
- Is http://localhost:3000/urdf/ur5e/meshes/base.dae reachable from the browser in which you open Studio in?
- Can you check the chrome developer tools and see if there are any failing network requests?

**Reply by MarcoGi89** [reply]
Thank you for the prompt reply.

- Yes, /robot_description topic is correctly visualized in Studio
- Opening another tab from that browser and navigating to http://localhost:3000/urdf/ur5e/meshes/base.dae for example the file is automatically downloaded, so I guess it's reachable
- Opening the chrome console log I noticed this error, that seems to be the problem:

![image](https://github.com/foxglove/community/assets/55974806/e34e9313-21f7-449b-be10-902791785435)

I tried to add Allow CORS: Access-Control-Allow-Origin extension to Chrome, but without success. 
Any suggestions?

Thanks

**Reply by achim-k** [reply]
You have to adapt your HTTP server which serves the mesh files to send the appropriate CORS header (e.g. `Access-Control-Allow-Origin: *`).  What HTTP server are you using? If you use the Python built in http server, have a look [here](https://stackoverflow.com/a/21957017).

**Reply by MarcoGi89** [reply, marked as answer]
Hi @achim-k,

as you said I fixed the problem adapting the HTTP server.

In my case I followed [this](https://medium.com/pareture/simple-docker-nginx-cors-reverse-proxy-5202a3445847) and everything worked.

Thank you very much for your tips!
